src/SprykerEco/Zed/Unzer/Business/Payment/Filter/AbstractUnzerPaymentMethodFilter.php

```diff
@@ -48,6 +49,35 @@ protected function isMarketplaceUnzerPaymentMethod(PaymentMethodTransfer $paymen
         return $this->isUnzerPaymentProvider($paymentMethodTransfer) && strpos($paymentMethodTransfer->getPaymentMethodKeyOrFail(), SharedUnzerConfig::PLATFORM_MARKETPLACE) !== false;
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\PaymentMethodsTransfer $paymentMethodsTransfer
+     * @param \Generated\Shared\Transfer\PaymentMethodTransfer $paymentMethodTransfer
+     *
+     * @return bool
+     */
+    protected function hasPrioritizedMarketplaceUnzerPaymentMethod(
+        PaymentMethodsTransfer $paymentMethodsTransfer,
+        PaymentMethodTransfer $paymentMethodTransfer
+    ): bool {
+        if ($this->isMarketplaceUnzerPaymentMethod($paymentMethodTransfer)) {
+            return false;
+        }
+
+        foreach ($paymentMethodsTransfer->getMethods() as $availablePaymentMethodTransfer) {
+            $marketplaceEquivalentPaymentMethodKey = str_replace(
+                SharedUnzerConfig::PAYMENT_PROVIDER_TYPE,
+                SharedUnzerConfig::PAYMENT_PROVIDER_TYPE . SharedUnzerConfig::PLATFORM_MARKETPLACE,
+                $paymentMethodTransfer->getPaymentMethodKeyOrFail(),
```

Comment:
the same here and at similar places
please check with @kkicheglovspryker

Reply:
Already used in method AbstractUnzerPaymentMethodFilter::isMarketplaceUnzerPaymentMethod() which is invoked in line 62. So should be fine ...